### Task1

Write a python code of a program that reads the values for the three sides x, y, and z of a triangle,and then calculates its area. The area is calculated as follows:

=====================================================

**Sample Input1:**\
7, 10, 5

**Sample Output1:**\
53.88877434122991

=====================================================

**Sample Input2:**\
7, 9, 12

**Sample Output2:**\
117.13240371477058


In [1]:
#1
import math
x=int(input('Enter 1st value:'))
y=int(input('Enter 2nd value:'))
z=int(input('Enter 3rd value:'))
s=(x+y+z)/2
Area=s*math.sqrt((s-x)*(s-y)*(s-z))
print('Area of triangle is:',format(Area,'.4f'))

Enter 1st value:7
Enter 2nd value:9
Enter 3rd value:12
Area of triangle is: 117.1324


Write Python code of a program that reads three numbers form user and print the greatest number
among three numbers.

In [ ]:
x=int(input('Enter 1st number:'))
y=int(input('Enter 2nd number:'))
z=int(input('Enter 3rd number:'))
if x>y and x>z:
    print(x,'is greatest')
elif y>x and y>z:
    print(y,'is greatest')
else:
    print(z,'is greatest')

### Task2

Write Python code of a program that reads three sides of a triangle and check whether the triangle is valid or not.

=====================================================

**Sample Input1:**\
7, 10, 5

**Sample Output1:**\
Valid triangle

=====================================================

**Sample Input2:**\
2, 2, 4

**Sample Output2:**\
Not a valid triangle

=====================================================

**Sample Input3:**\
7, 9, 12

**Sample Output3:**\
Valid triangle


In [3]:
#2
x=float(input('Enter 1st number:'))
y=float(input('Enter 2nd number:'))
z=float(input('Enter 3rd number:'))
if x+y>z and y+z>x and z+x>y:
    print('Valid triangle')
else:
    print('Not a valid triangle')

Enter 1st number:2
Enter 2nd number:2
Enter 3rd number:4
Not a valid triangle


### Task3

Write Python code of a program that reads a number as a year and determines whether it is a leap
year or not.\
Three conditions are used to identify leap years:\
● The year can be evenly divided by 4, is a leap year, unless\
● The year can be evenly divided by 100, it is NOT a leap year, unless:\
● The year is also evenly divisible by 400. Then it is a leap year.\
[which implies - if the year can be divided by 4 and NOT divided by 100, then it’s a leap year.
If the year can be divided by 4, divided by 100, and divided by 400, then it’s a leap year.]

For example, the years 2000, 2024, 2028, and 2400 are leap years, while 1800, 1900, 2100, 2200,
2300, and 2500 are NOT leap years.

Here, 2000, 2400 are divided by 4, 100, and 400. Thus leap year.\
Here, 2024, 2028 are divided by 4 but not divided by 100. Thus leap year

In [5]:
#3
# Python program to check if year is a leap year or not

year =int(input('Enter year:'))

if (year % 4) == 0:
    if (year % 100) == 0:
        if (year % 400) == 0:
            print("{} is a leap year".format(year))
        else:
            print("{} is not a leap year".format(year))
    else:
        print("{} is a leap year".format(year))
else:
    print("{} is not a leap year".format(year))

Enter year:2024
2024 is a leap year


### Task4

Write a Python program that reads a number and finds the sum of the series of 1 +11 + 111 + 1111+ ….+N terms.

=====================================================

**Sample Input1:**\
5

**Sample Output1:**\
1 + 11 + 111 + 1111 + 11111

The Sum is: 1234

=====================================================

**Sample Input2:**\
8

**Sample Output2:**\
1 + 11 + 111 + 1111 + 11111 + 111111 + 1111111 + 11111111\
The Sum is: 12345678

In [9]:
#4
numb=int(input('Enter a number:'))
sum = 0
j = 1
     
for i in range(1, numb + 1):
    sum = sum + j
    j = (j * 10) + 1
print('Sum of 1+11+111+...+N terms is:',sum)

Enter a number:8
Sum of 1+11+111+...+N terms is: 12345678


### Task5

Write a Python program that reads a number and displays the multiplication table of the given
integer.

**Sample Input:** 15

**Sample Output :**\
15X1=15\
15X2=30\
...\
...\
15X10=150

In [10]:
#5
num = int(input(" Enter the number : "))    
# using for loop to iterate multiplication 10 times     
print("Multiplication Table of : ")  
for i in range(1,11):    
    print(num,'x',i,'=',num*i)

 Enter the number : 10
Multiplication Table of : 
10 x 1 = 10
10 x 2 = 20
10 x 3 = 30
10 x 4 = 40
10 x 5 = 50
10 x 6 = 60
10 x 7 = 70
10 x 8 = 80
10 x 9 = 90
10 x 10 = 100


### Task9

Write a python program that prints alphabet pattern 'Z' of size N using * where N will be given as input.

In [16]:
#9
numb=int(input('Enter size:'))
i=1
j=numb-2
for row in range(numb):
    for col in range(numb):
        if (row==0 or row==numb-1):
            print('*',end='')
        elif col==j and row==i:
            print('*',end='')
            j=j-1
            i=i+1
        else:
            print(end=' ')
    print()

Enter size:5
*****
   * 
  *  
 *   
*****


### Task6

Write a python program that takes integer inputs from the user until the user gives "Stop".
Print average and product of all numbers

In [12]:
#6
list1=[]
sum=0
product=1
print('Type "Stop" to end this code')
while True:
    numb=input('Enter numbers you want:')
    if numb=='Stop':
        break
    list1.append(numb)
for char in list1:
    sum=sum+int(char)
    product=product*int(char)
print('Average is {}'.format(sum/len(list1)))
print('Product is {}'.format(product))

Type "Stop" to end this code
Enter numbers you want:1
Enter numbers you want:2
Enter numbers you want:3
Enter numbers you want:4
Enter numbers you want:5
Enter numbers you want:Stop
Average is 3.0
Product is 120


### Task7

Write a Python program to count the total number of alphabets, digits, and special characters in a string. Here, space is considered as a special character.

====================================================

**Sample Input:**\
Welcome to geeksforgeeks.com

**Sample Output:**\
The number of Alphabets in the string is: 25\
The number of Digits in the string is : 0\
The number of Special characters in the string is : 3

In [13]:
#7
# Python program to Count Alphabets Digits and Special Characters in a String
 
string = input("Please Enter your Own String : ")
alphabets = digits = special = 0

for i in range(len(string)):
    if(string[i].isalpha()):
        alphabets = alphabets + 1
    elif(string[i].isdigit()):
        digits = digits + 1
    else:
        special = special + 1
        
print("\nTotal Number of Alphabets in this String :  ", alphabets)
print("Total Number of Digits in this String :  ", digits)
print("Total Number of Special Characters in this String :  ", special)

Please Enter your Own String : Welcome to geeksforgeeks.com

Total Number of Alphabets in this String :   25
Total Number of Digits in this String :   0
Total Number of Special Characters in this String :   3


### Task8

Write a Python program to find the largest and smallest word in a string.

**Sample Input:**\
It is a string with the smallest and largest word.

**Sample Output:**\
The largest word is “smallest” and the smallest word is 'a'.

In [15]:
#8
Line='It is a string with the smallest and largest word '

words=[x for x in Line.split()]
sorted_words = sorted(words, key= lambda y: len(y), reverse = True)
longest_word = sorted_words[0]
shortest_word = sorted_words[-1]

print('In the string:', end=' ')
for j in words:
    print(j, end=' ')
print()
print('Longest word is:',longest_word)
print('and smallest word is:',shortest_word)

In the string: It is a string with the smallest and largest word 
Longest word is: smallest
and smallest word is: a


### Task10

Write a Python program that takes two strings from the user. Then remove characters from the first string which are present in the second string. [Avoid built-in function]

=====================================================

**Sample Input1:**\
India is great\
is

**Sample Output1:**\
nda great

=====================================================

**Sample Input2:**\
this is cSe110 course and we love it

**Sample Output2:**\
ths s cse0 curse nd we e t

In [18]:
#10
n1=input('Enter first string:')
n2=input('Enter second string:')
a=''

for i in n1:
    if i not in n2:
        a=a+i
print(a)

Enter first string:india is great
Enter second string:is
nda  great


### Task11

Write a Python program to Capitalize the first character of each word in a String [You cannot use
the built-in upper() function]

**Sample Input:**\
I love python programming

**Sample Output:**\
I Love Python Programming

In [21]:
#11
list1=input('Enter digits separated by space:')
list2=list1.split(' ')
for i in list2:
    if ord(i[0])>=97 and ord(i[0])<=122:
        new=i.replace(i[0],chr(ord(i[0])-32))
    print(new,end=' ')

Enter digits separated by space:i love python programming
I Love Python Programming 

### Task12

Write a Python program that takes a list of numbers separated by comma(“,”) as input of a List
A(Without using a loop). Multiply the numbers and print the product. [Hint: You can use split()]

=====================================================

**Sample Input1:**\
1,2,3,4,5

**Sample output1:**\
Product:120

=====================================================

**Sample Input2:**\
10,0,20,3,1

**Sample Output2:**\
Product:0

In [23]:
#12
list1=input('')
list2=list1.split(',')
product=1
for i in list2:
    product=product*int(i)
print('Product:',product)

1,2,3,4,5
Product: 120


### Task13

Write a Python program that takes two lists from the user. Print both lists simultaneously such that list1 should display items in original order and list2 in reverse order.

**Sample Input:**\
List1 = [10,20,30,40]\
List2 = [100,200,300,400]

**Sample Output:**\
10 400\
20 30\
30 200\
40 100

In [24]:
#13
list1=input('')
list2=input('')
list3=list1.split(' ')
list4=list2.split(' ')
list5=list4[-1::-1]
for i in range(len(list3)):
    print(list3[i],list5[i])

1 2 3 4 5
6 7 8 9 10
1 10
2 9
3 8
4 7
5 6


### Task14

Write a Python program to reverse strings in a given list of string values without using reverse() or reversed().

**Sample Input:** ['Red','Green','Blue','White','Black']

**Sample Output:** ['deR', 'neerG', 'eulB', 'etihW', 'kcalB']

In [25]:
#14
list1=input('')
list2=list1.split(',')
a=''
b=[]
for i in list2:
    for j in range(len(i)-1,-1,-1):
        a=a+i[j]
    b.append(a)
    a=''
print(b)

red,green,blue,white,black
['der', 'neerg', 'eulb', 'etihw', 'kcalb']


### Task15

Write a Python program that takes a list from the user. Then find the first even and first odd
number in the list.

=====================================================

**Sample Input1:**\
[1, 3, 5, 7, 4, 1, 6, 8]

**Sample Output1:**\
First even:4 and First odd:1

=====================================================

**Sample Input2:**\
4\
[55, 5, 7, 9, 11]

**Sample Output2:**\
First even: not found and First odd: 55

=====================================================

**Sample Input3:**\
[50, 52, 6, 2, 10]

**Sample Output3:**\
First even: 50 and First odd: not found

In [26]:
#15
list1=input('')
list2=list1.split(',')
eve=[]
od=[]
for i in list2:
    if int(i)%2==0:
        eve.append(i)
    else:
        od.append(i)
print('First even is:',eve[0])
print('First odd is:',od[0])

14,112,16,1
First even is: 14
First odd is: 1


### Task16

Write a Python program that takes a list A and a number as input from the user. Create a list by
concatenating the input list which range goes from 1 to n.

=====================================================

**Sample input1:**\
A = ['p', 'q']\
N = 5

**Sample Output1:**\
['p1', 'q1', 'p2', 'q2', 'p3', 'q3', 'p4', 'q4', 'p5', 'q5']

=====================================================

**Sample input2:**\
A = ['p', 'q', 'r']\
N = 3

**Sample Output2:**\
['p1', 'q1', 'r1', 'p2', 'q2', 'r2', 'p3', 'q3', 'r3']

=====================================================

**Sample input3:**\
A = ['p', 'q', 'r','s']\
N = 2

**Sample Output3:**\
['p1', 'q1', 'r1', 's1', 'p2', 'q2', 'r2', 's2']


In [27]:
#16
list1=input('')
list2=list1.split(',')
a=[]
numb=int(input(''))
for i in range(1,numb+1):
    for j in list2:
        a.append(j+str(i))
print(a)

1,2,3,4
4
['11', '21', '31', '41', '12', '22', '32', '42', '13', '23', '33', '43', '14', '24', '34', '44']


𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻 2:
Write a Python program that takes a word as an input from the user and does 
the following. While printing, use a space after each letter or ASCII value.

If the word has an even number of letters, then print the middle two 
letters replaced with their ASCII values.
If the word has an odd number of letters, then  print the middle three 
letters replaced with their ASCII values.
If the word is less than four letters long, simply print the word itself.

[Do not use any built-in functions of String class except len() and ord()]

=========================================================

**Sample Input 1:**\
computer

**Sample Output 1:**\
c o m 112 117 t e r

**Explanation 1:**\
The word 'computer' has 8 letters which is an even number and the length is 
not less than 4. Hence, the middle 2 letters 'p' and 'u' are replaced with 
the ASCII values of 'p' and 'u' which are 112 and 117. Notice the spaces 
after each letter or ASCII value in the output.

=========================================================

**Sample Input 2:**\
programming

**Sample Output 2:**\
p r o g 114 97 109 m i n g

**Explanation 2:**\
The word 'programming' has 11 letters which is an odd number and the length 
is not less than 4. Hence, the middle 3 letters 'r', 'a' and 'm' are 
replaced with the ASCII values of 'r', 'a', and 'm' which are 114 and 97 
and 109. Notice the spaces after each letter or ASCII value in the output.

=========================================================

**Sample Input 3:**\
mid

**Sample Output 3:**\
mid

**Explanation 3:**\
The word 'mid' has 3 letters which is an odd number. But the length is less 
than 4. So, the original word was printed.

In [6]:
n=input('')
if len(n)<4:
    print(n)
else:

    if len(n)%2==0:
        ev1=((len(n))/2)-1
        ev2=(len(n))/2
        for i in range(len(n)):
            ev1=((len(n))/2)-1
            ev2=(len(n))/2
            if ev1<=i<=ev2:
                print(ord(n[i]),end=' ')
            else:
                print(n[i],end=' ')
    else:
        for j in range(len(n)):
            as1=((len(n)-1)/2)-1
            as2=((len(n)-1)/2)+1
            if as1<=j<=as2:
                print(ord(n[j]),end=' ')
            else:
                print(n[j],end=' ')

programming
p r o g 114 97 109 m i n g 

𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻 1:
Write a python program that takes two numbers (Integer) as inputs from the 
user. Consider the 1st number to be the total number of rows and the 2nd 
number to be the total number of columns of a rectangle. [For this 
question, you can assume the second input is always greater than the first 
input. So, no checking is needed for ensuring this condition.] Finally, 
print a rectangle as shown in the output.
=============================================================
Sample Input 1:
3
6

Sample Output 1:
1=====
12====
123===
Explanation 01:
3 is the total number of rows and  6 is the total number of columns. For 
each row, numbers starting from 1 to that row number are printed and the 
rest of the columns are filled by equal 😊) to sign.
=============================================================
Sample Input 2:
6
7
Sample Output 2:
1======
12=====
123====
1234===
12345==
123456=
Explanation 02:
6, is the total number of rows, and 7 is the total number of columns. For 
each row, numbers starting from 1 to that row number are printed and the 
rest of the columns are filled by equal 😊) to sign.



𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻 2:
Take a lowercase word as an input from the user. Modify the word in such a 
way that the letters in the multiple of 3 positions become uppercase except 
position 0. Lastly, print the newly modified word.

[You are not allowed to use upper() method]
=============================================================
Sample Input1:
jupyter
Sample Output1:
jupYteR
Explanation 1:
The characters ‘y’ and ‘r’ are in positions 3 and 6 respectively. So, these 
two characters will be converted to uppercase in the modified string.
=============================================================
Sample Input2:
python notebook
Sample Output2:
pytHon noTebOok
Explanation 2:
The characters ‘h’, ‘t’, and ‘o’ are in positions 3, 9, and 12 
respectively. So, these three characters will be converted to uppercase in 
the modified string.

𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻 1:
You will be given three inputs M, N, and P. The M indicates the number of 
chocolates you have. And there is a shop where you can get P new chocolates 
if you return N empty chocolate packets.
You need to find the total number of chocolates that you can have and print 
it.
================================
Sample Input 1:
100
4
1
Sample Output 1:
133

Explanation 1:
In this input, the shop will return 1 new chocolate for every 4 empty 
packets. So if you have 100 chocolates in the beginning, you can have more 
chocolates through the following steps.

Eat 100 chocolates -> 100 empty packs.
100 empty packs -> 25 new chocolates.
Eat 25 chocolates -> 25 empty packets.
25 empty packets -> 6 new chocolates and 1 empty packet.
Eat 6 chocolates -> (6 + 1 remaining) = 7 empty packs.
7 empty packets -> 1 new chocolate and 3 empty packets.
Eat 1 chocolate -> (1 + 3 remaining)  = 4 empty packs.
4 empty packs -> 1 new chocolate.
Eat 1 chocolate -> 1 empty packet.

No new chocolate can be obtained using only 1 packet.
Now counting all the chocolates you have eaten you’ll get
(100 + 25 + 6 + 1 + 1) = 133 chocolates in total. So, the output is 133.

================================
Sample Input 2:
100
10
3
Sample Output 2:
139

Explanation 2:
In this input, the shop will return 3 new chocolates for every 10 empty 
packets. So if you have 100 chocolates in the beginning, you can have more 
chocolates through the following steps.

Eat 100 chocolates -> 100 empty packs.
100 empty packets -> 30 new chocolates. (3 for each 10)
Eat 30 chocolates -> 30 empty packs.
30 empty packs -> 9 new chocolates.(3 for each 10)
Eat 9 chocolates -> 9 empty packs.
You can’t get any more new chocolates with 9 empty packs since 9 is less 
than 10.
Now counting all the chocolates you have eaten. You’ll get
(100 + 30 + 9) = 139 chocolates in total. So, the output is 139.

𝗤𝘂𝗲𝘀𝘁𝗶𝗼𝗻 2:
Write a Python program that takes a word as an input from the user and does 
the following. While printing, use a space after each letter or ASCII value.

If the word has an even number of letters, then print the middle two 
letters replaced with their ASCII values.
If the word has an odd number of letters, then  print the middle three 
letters replaced with their ASCII values
If the word is less than four letters long, simply print the word itself.

[Do not use any built-in functions of String class except len() and ord()]
=========================================================
Sample Input 1:
computer

Sample Output 1:
c o m 112 117 t e r

Explanation 1:
The word 'computer' has 8 letters which is an even number and the length is 
not less than 4. Hence, the middle 2 letters 'p' and 'u' are replaced with 
the ASCII values of 'p' and 'u' which are 112 and 117. Notice the spaces 
after each letter or ASCII value in the output.

=========================================================
Sample Input 2:
programming

Sample Output 2:
p r o g 114 97 109 m i n g

Explanation 2:
The word 'programming' has 11 letters which is an odd number and the length 
is not less than 4. Hence, the middle 3 letters 'r', 'a' and 'm' are 
replaced with the ASCII values of 'r', 'a', and 'm' which are 114 and 97 
and 109. Notice the spaces after each letter or ASCII value in the output.
=========================================================
Sample Input 3:
mid

Sample Output 3:
mid

Explanation 3:
The word 'mid' has 3 letters which is an odd number. But the length is less 
than 4. So, the original word was printed.

Write a python program that asks the user for a range (a starting number(inclusive) and an ending number(inclusive)). Then take another input to check, if the product of all the digits of each number is divisible by this input or not. If it is divisible, then print it.

Sample Input 1:
25
30
4
Sample Output 1:
12 16 0

Explanation 1:
The product of all the digits of each number from 25 to 30 is 2x5, 2x6, 2x7, 2x8, 2x9, 3x0. The final product is 10,12,14,16,18,0 respectively. Out of these numbers only 12, 16 and 0 is divisible by 4 and therefore it is printed.

Sample Input 2:
351
356
9
Sample output 2:
45 90

In [1]:
n=input('Enter start and end using comma:')
list1=n.split(',')
list2=[]
list3=[]
list4=[]
for i in range(int(list1[0]),int(list1[-1])+1):
    list2.append(str(i))
for j in list2:
    multi=int(j[0])*int(j[-1])
    list3.append(multi)
for k in list3:
    if k%4==0:
        list4.append(k)
    else:
        continue

print(list4)

Enter start and end using comma:25,30
[12, 16, 0]


Write a python code which will read a file (goal stats.txt) that conatins player names, their club names and total goals they have scored this season.(Separated by '|')
Your task is to then create a dictionary with only player names and the goals they have scored. Then, output a file where all the players name will be listed sorted by their scored goals. The highest scorer will be top most line.